### PR TITLE
[#2228] Viber source

### LINKED ISSUE #2228: Viber Source
**Is your feature request related to a problem? Please describe.**

As an Airy user I want to be able to connect a new source of type `viber` to my Airy instance, 
consuming events from it and sending back messages,
and being able to look at conversations of type `viber` in the Inbox UI.

**Describe the solution you'd like**

- Store credentials in `airy.yaml` configuration for source - or on channel level (comparable to page tokens)

- Requests are authenticated via header `X-Viber-Auth-Token`: https://developers.viber.com/docs/api/rest-bot-api/#authentication-token

- Webhooks to subscribe for: https://developers.viber.com/docs/api/rest-bot-api/#setting-a-webhook

Conversation Started Event (https://developers.viber.com/docs/api/rest-bot-api/#conversation-started):
```
{
   "event":"conversation_started",
   "timestamp":1457764197627,
   "message_token":4912661846655238145,
   "type":"open",
   "context":"context information",
   "user":{
      "id":"01234567890A=",
      "name":"John McClane",
      "avatar":"http://avatar.example.com",
      "country":"UK",
      "language":"en",
      "api_version":1
   },
   "subscribed":false
}
```
This event needs to be responded with HTTP 200 + optional: a Welcome Message in the body (https://developers.viber.com/docs/api/rest-bot-api/#sending-a-welcome-message)

The welcome message should be used to convert the user to send a message which subscribes the user, otherwise the Send Message via API route can't be used to reach users that haven't subscribed yet.

Receive Message from User Event (https://developers.viber.com/docs/api/rest-bot-api/#receive-message-from-user):
```
{
   "event":"message",
   "timestamp":1457764197627,
   "message_token":4912661846655238145,
   "sender":{
      "id":"01234567890A=",
      "name":"John McClane",
      "avatar":"http://avatar.example.com",
      "country":"UK",
      "language":"en",
      "api_version":1
   },
   "message":{
      "type":"text",
      "text":"a message to the service",
      "media":"http://example.com",
      "location":{
         "lat":50.76891,
         "lon":6.11499
      },
      "tracking_data":"tracking data"
   }
}
```

Subscribed Event (https://developers.viber.com/docs/api/rest-bot-api/#subscribed) - Requirement to send messages to a User:

```
{
   "event":"subscribed",
   "timestamp":1457764197627,
   "user":{
      "id":"01234567890A=",
      "name":"John McClane",
      "avatar":"http://avatar.example.com",
      "country":"UK",
      "language":"en",
      "api_version":1
   },
   "message_token":4912661846655238145
}
```

Corresponding Event: https://developers.viber.com/docs/api/rest-bot-api/#unsubscribed

Message Receipts: https://developers.viber.com/docs/api/rest-bot-api/#message-receipts-callbacks

APIs to send to: 

- Send Message via API: https://developers.viber.com/docs/api/rest-bot-api/#send-message

There is a hourly rate limit of sending 100 individual messages per recipient without responses via the API. 

Alternatively, Broadcast Messages can be send to subscribers via Broadcast Message via API (https://developers.viber.com/docs/api/rest-bot-api/#broadcast-message).

The Response  for sending a message contains a `message_token` property which should be tracked for following `delivered` and `seen` events: 
```
{
  "status": 0,
  "status_message": "ok",
  "message_token": 5602570000333709000,
  "chat_hostname": "SN-CHAT-24_"
}
```
Delivered Event:
```
{
  "event": "delivered",
  "timestamp": 1627814555340,
  "chat_hostname": "SN-CHAT-25_",
  "message_token": 5602570000333709000,
  "user_id": "BtrgXI7x1u6hYXaOi+GQyw=="
}
```
Seen Event:
```
{
  "event": "seen",
  "timestamp": 1627814556898,
  "chat_hostname": "SN-CHAT-25_",
  "message_token": 5602570000333709000,
  "user_id": "BtrgXI7x1u6hYXaOi+GQyw=="
}
````

The metadata status of the messages and conversations should be updated accordingly.

**Describe alternatives you've considered**
-

**Additional context**
-

### PR DESCRIPTION
**Pull request by chrismatix**
resolves #2228 

- [x] Docs!!
- [x] Webhook with welcome message
- [x] (Dis-) connect channels
- [x] Map delivery and seen state
- [x] ingest messages
- [x] send messages

### CODE REVIEW

File: res.yaml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    core.airy.co/prometheus: spring
+spec:
+  ports:
+    - name: web
+      port: 80
+      targetPort: 8080
+      protocol: TCP
+  type: NodePort
+  selector:
+    app: sources-viber-connector
```
Comment (ljupcovangelski) [inline res.yaml:125]: Why is this file needed?

File: lib/typescript/render/providers/chatplugin/ChatPluginRender.tsx [hunk 2]
```diff
@@ -60,7 +58,7 @@ function render(content: ContentUnion, props: RenderPropsUnion) {
   }
 }
 
-function mapContent(message: Message): ContentUnion {
+function mapContent(message): ContentUnion {
```
Comment (AitorAlgorta) [inline lib/typescript/render/providers/chatplugin/ChatPluginRender.tsx:61]: 👀

File: lib/typescript/render/providers/chatplugin/components/RichText/index.tsx [hunk 3]
```diff
@@ -1,22 +1,19 @@
...
   fallback: string;
   containsRichText: boolean;
 };
 
 export const RichText = (props: RichTextRenderProps) => {
-  const {message, text, fromContact} = props;
+  const {text, fromContact} = props;
   return (
-    <div className={fromContact ? styles.contactContent : styles.memberContent} id={`message-item-${message.id}`}>
+    <div className={fromContact ? styles.contactContent : styles.memberContent}>
```
Comment (AitorAlgorta) [inline lib/typescript/render/providers/chatplugin/components/RichText/index.tsx:16]: Is the cypress test still passing?

**Review by AitorAlgorta — APPROVED** [review]
LGTM just some questions

File: lib/typescript/render/providers/chatplugin/components/RichText/index.tsx (hunk 3 above)
Comment (chrismatix) [reply lib/typescript/render/providers/chatplugin/components/RichText/index.tsx:16]: This id was a [duplicate](https://github.com/airyhq/airy/blob/3aa1f97fbe5137319ca43b4b6c64574b76185d57/frontend/ui/src/pages/Inbox/Messenger/MessageList/index.tsx#L172)

File: lib/typescript/render/providers/chatplugin/ChatPluginRender.tsx (hunk 2 above)
Comment (chrismatix) [reply lib/typescript/render/providers/chatplugin/ChatPluginRender.tsx:61]: Yes this was not accurate: We use the renderer for templates and suggested replies also, which are **not** messages. The actual interface is:
```typescript
interface message {
  content: any;
  fromContact?: bool
}
```

**Review by ljupcovangelski — APPROVED** [review]
Great work :+1:

File: lib/typescript/render/providers/chatplugin/ChatPluginRender.tsx (hunk 2 above)
Comment (chrismatix) [reply lib/typescript/render/providers/chatplugin/ChatPluginRender.tsx:61]: Therefore I removed the explicit dependency on message cc @airyhq/codeowners-frontend

**Review by ljupcovangelski — APPROVED** [review]
